File: Lab/Lab_1_2/src/Lab_2/BinarySearch.py
def search(arr, start, end, key):
    if (start < end):
        mid = start + (end - start) // 2

        if (key == arr[mid]):
            return mid
        elif key > arr[mid]:
            return search(arr, mid + 1, end, key)
        else:
            return search(arr, start, mid, key)
    return -1


array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

result = search(array, 0, len(array) - 1, 9)
print(result)

File: Lab/Lab_1_1/src/main/java/ip/group/Lab_2/BinarySearch.java
package ip.group.Lab_2;

public class BinarySearch {

    public static int binary(int arr[], int start, int end, int key) {
        if (start < end) {
            int mid = start + (end - start) / 2;
            if (arr[mid] == key)
                return mid;
            else if (arr[mid] > key)
                return binary(arr, start, mid, key);
            else
                return binary(arr, mid + 1, end, key);
        }
        return -1;
    }

    public static void main(String[] args) {

        BinarySearch binarySearch = new BinarySearch();

        int arr[] = {0, 2, 3, 4, 5, 7, 8, 9};

        int result = binarySearch.binary(arr, 0, arr.length, 0);
        System.out.println(result);
    }
}